core/capabilities/ccip/ccipsolana/msghasher.go

```diff
@@ -20,12 +20,14 @@ import (
 // Compatible with:
 // - "OnRamp 1.6.0-dev"
 type MessageHasherV1 struct {
-	lggr logger.Logger
+	lggr           logger.Logger
+	extraDataCodec cciptypes.ExtraDataCodec
```

Comment:
Lets get rid of the public interface too
```suggestion
	extraDataCodec ccipcommon.NewExtraDataCodec
```

Reply:
I assume Will meant "extraDataCodec ccipcommon.ExtraDataCodec". Use the type from this repo itself.

Reply:
This is also because of the mocking, `cciptypes.ExtraDataCodec` interface gives more flexibility

Reply:
Hmm, if we are going to remove the one from chainlink-ccip repo, perhaps just create it inside ccipcommon in this repo

Reply:
Oh true